projects/fal/src/fal/app.py

```diff
@@ -36,6 +40,61 @@ async def _call_any_fn(fn, *args, **kwargs):
         return fn(*args, **kwargs)
 
 
+@dataclass
+class IsolateChannel:
+    address: str
+    _stack: AsyncExitStack | None = field(repr=False, default=None)
+
+    async def __aenter__(self) -> async_grpc.Channel:
+        self._stack = AsyncExitStack()
+        channel = await self._stack.enter_async_context(
+            async_grpc.insecure_channel(
+                self.address,
+                options=[
+                    ("grpc.max_send_message_length", -1),
+                    ("grpc.max_receive_message_length", -1),
+                    ("grpc.min_reconnect_backoff_ms", 0),
+                    ("grpc.max_reconnect_backoff_ms", 100),
+                    ("grpc.dns_min_time_between_resolutions_ms", 100),
+                ],
+            )
+        )
+        channel_status = channel.channel_ready()
+        try:
+            await asyncio.wait_for(channel_status, timeout=60)
+        except asyncio.TimeoutError:
+            raise Exception("Timed out trying to connect to local isolate")
+        else:
+            return channel
+
+    async def __aexit__(self, *args):
+        if self._stack is None:
+            return None
+
+        try:
+            await self._stack.aclose()
+        finally:
+            self._client = None
+            self._stack = None
+
+
+async def _set_logger_labels(logger_labels: dict[str, str]):
+    grpc_port = os.environ.get("NOMAD_ALLOC_PORT_grpc")
+
+    try:
+        async with IsolateChannel(f"localhost:{grpc_port}") as channel:
+            isolate = definitions.IsolateStub(channel)
```

Comment:
can we cache this connection? we should avoid opnening a grpc connection each time